src/elements/content-sharing/hooks/useContactsByEmail.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import * as React from 'react';
+import noop from 'lodash/noop';
+import API from '../../../api';
+import type { UserCollection, UserMini } from '../../../common/types/core';
+import type { ContactByEmailObject, ContentSharingHooksOptions, GetContactsByEmailFnType } from '../types';
+
+/**
+ * Generate the getContactsByEmail() function, which is used for looking up contacts added to the collaborators field in the USM.
+ *
+ * @param {API} api
+ * @param {string} itemID
+ * @param {ContentSharingHooksOptions} options
+ * @returns {GetContactsByEmailFnType | null}
+ */
+function useContactsByEmail(
+    api: API,
+    itemID: string,
+    options: ContentSharingHooksOptions,
+): GetContactsByEmailFnType | null {
+    const [getContactsByEmail, setGetContactsByEmail] = React.useState<null | GetContactsByEmailFnType>(null);
+    const { handleSuccess = noop, handleError = noop, transformUsers } = options;
+
+    React.useEffect(() => {
+        if (getContactsByEmail) return;
+
+        const resolveAPICall = (
+            resolve: (result: ContactByEmailObject | Array<UserMini>) => void,
+            response: UserCollection,
+            transformFn: ?Function,
+        ) => {
+            handleSuccess(response);
+            // A successful API call will always return an entries array, but we still need these checks for Flow purposes
+            if (response && response.entries && response.entries.length) {
+                return resolve(transformFn ? transformFn(response) : response.entries);
+            }
+            return resolve({});
+        };
+
+        const updatedGetContactsByEmailFn: GetContactsByEmailFnType = () => (filterTerm: {
+            [emails: string]: string,
+        }) => {
+            if (!filterTerm || !filterTerm.emails || !Array.isArray(filterTerm.emails) || !filterTerm.emails.length) {
```

Comment:
Can you can get rid of `!filterTerm.emails` and let `Array.isArray` handle the null/undefined check?